src/Client.js

```diff
@@ -5,22 +5,79 @@ import type {
   JSONValue,
   LandRequest,
   Persona,
-  StatusEvent
+  StatusEvent,
+  Settings,
+  PullRequest,
+  BuildStatus
 } from './types';
+import Logger from './Logger';
 
 export default class Client {
   hostAdaptor: Host;
   ciAdaptor: CI;
   persona: Persona;
+  serverSettings: Settings;
 
-  constructor(host: Host, ci: CI, persona: Persona) {
+  constructor(host: Host, ci: CI, persona: Persona, settings: Settings) {
     this.hostAdaptor = host;
     this.ciAdaptor = ci;
     this.persona = persona;
+    this.serverSettings = settings;
   }
 
   async isAllowedToLand(pullRequestId: string) {
-    return await this.hostAdaptor.isAllowedToLand(pullRequestId);
```

Comment:
This was one of the main things.
Moving the "isAllowedToLand" logic out of the HostAdaptor.

This is much better